### PR TITLE
Test expiration header for lc rules with tags

### PR DESCRIPTION
**Pull request by ofriedma**
Signed-off-by: Or Friedmann <ofriedma@redhat.com>

### CODE REVIEW

**Comment by mattbenjamin** [comment]
I think I already submitted a fix for the first issue, so...

**Comment by mattbenjamin** [comment]
maybe just send the header tag test?

**Comment by mattbenjamin** [comment]
ok, never mind, don't see it

File: s3tests_boto3/functional/test_s3.py [hunk 1]
```diff
@@ -9206,6 +9210,116 @@ def test_lifecycle_expiration_header_head():
...
+            "Expiration": {
+                "Days": 1
+            },
+            "ID": "rule1"
+            },
+        ]
+    }
+    response = client.put_bucket_lifecycle_configuration(
+        Bucket=bucket_name, LifecycleConfiguration=lifecycle)
+    #response = client.put_object(Bucket=bucket_name, Key=key1, Body=body1)
```
Comment (mattbenjamin) [inline s3tests_boto3/functional/test_s3.py:9266]: what are these commented lines doing?

Comment (ofriedma) [reply s3tests_boto3/functional/test_s3.py]: Fixed

**Comment by ofriedma** [comment]
@mattbenjamin can you take a look please?

File: s3tests_boto3/functional/test_s3.py [hunk 2]
```diff
@@ -9412,20 +9413,18 @@ def setup_lifecycle_expiration(client, bucket_name, rule_id, delta_days,
 
 def check_lifecycle_expiration_header(response, start_time, rule_id,
                                       delta_days):
-    print(response)
-    print(response['ResponseMetadata']['HTTPHeaders'])
+    expr_exists = ('x-amz-expiration' in response['ResponseMetadata']['HTTPHeaders'])
+    if (not expr_exists):
+        return False
+    expr_hdr = response['ResponseMetadata']['HTTPHeaders']['x-amz-expiration']
```
Comment (mattbenjamin) [inline s3tests_boto3/functional/test_s3.py:9419]: I'm trying to remember, can't these throw an (unhandled) exception?

File: s3tests_boto3/functional/test_s3.py [hunk 3]
```diff
@@ -9412,20 +9413,18 @@ def setup_lifecycle_expiration(client, bucket_name, rule_id, delta_days,
...
+    m = re.search(r'expiry-date="(.+)", rule-id="(.+)"', expr_hdr)
 
-    eq(exp_diff.days, delta_days)
-    eq(m.group(2), rule_id)
+    expiration = dateutil.parser.parse(m.group(1))
+    days_to_expire = ((expiration.replace(tzinfo=None) - start_time).days == delta_days)
+    rule_eq_id = (m.group(2) == rule_id)
 
-    return True
+    return  days_to_expire and rule_eq_id
```
Comment (mattbenjamin) [inline s3tests_boto3/functional/test_s3.py:9427]: ok

**Review by mattbenjamin — COMMENTED** [review]
one question, looks good

File: s3tests_boto3/functional/test_s3.py (hunk 2 above)
Comment (ofriedma) [reply s3tests_boto3/functional/test_s3.py:9419]: from the code it looks like the tests are expecting to get bool and to compare it to True or False

For example:

eq(check_lifecycle_expiration_header(response, now, 'rule1', 1), True)

Comment (mattbenjamin) [reply s3tests_boto3/functional/test_s3.py:9419]: oh, ok, you checked for x-amz-expiration on line 9416

**Review by mattbenjamin — APPROVED** [review]
lgtm

**Comment by mattbenjamin** [comment]
@ofriedma how will we do the ci/test?  per 1x1, the plan is to use qa suite rathr than s3-tests?

**Pull request merged**